src/main/java/com/metamx/common/CompressionUtils.java

```diff
@@ -99,6 +99,7 @@ public static long zip(File directory, OutputStream out) throws IOException
         totalSize += Files.asByteSource(file).copyTo(zipOut);
       }
       zipOut.closeEntry();
+      zipOut.flush();
```

Comment:
I think it should be left to the caller to handle exceptions thrown when closing the outputstream and not rely on this method to flush.